File: app/src/main/java/opencrowd/hgc/hgcwallet/ui/customviews/TitleBarWrapper.java
package opencrowd.hgc.hgcwallet.ui.customviews;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import opencrowd.hgc.hgcwallet.R;

public class TitleBarWrapper {
    View mRootView;
    TextView mTextView;
    ImageView mCloseButton;

    public  TitleBarWrapper(View rootView) {
        mRootView = rootView;
        mTextView = (TextView)rootView.findViewById(R.id.text_new_wallet);
        mCloseButton = (ImageView) rootView.findViewById(R.id.image_close);
    }

    public void setTitle(String title) {
        mTextView.setText(title);
    }

    public void setTitle(int resId) {
        mTextView.setText(resId);
    }

    public void setImageResource(int image) {mCloseButton.setImageResource(image);}



    public void setCloseButtonHidden(Boolean hidden) {
        mCloseButton.setVisibility(hidden ? View.INVISIBLE : View.VISIBLE);
    }

    public void setOnCloseButtonClickListener(View.OnClickListener listener) {
        mCloseButton.setOnClickListener(listener);
    }
}

File: app/src/main/java/hedera/hgc/hgcwallet/ui/customviews/TitleBarWrapper.kt
package hedera.hgc.hgcwallet.ui.customviews

import android.view.View
import android.widget.ImageView
import android.widget.TextView

import hedera.hgc.hgcwallet.R

class TitleBarWrapper(mRootView: View) {
    private val mTextView: TextView
    private val mCloseButton: ImageView

    init {
        mTextView = mRootView.findViewById<TextView>(R.id.text_new_wallet)
        mCloseButton = mRootView.findViewById<ImageView>(R.id.image_close)
    }

    fun setTitle(title: String) {
        mTextView.text = title
    }

    fun setTitle(resId: Int) {
        mTextView.setText(resId)
    }

    fun setImageResource(image: Int) {
        mCloseButton.setImageResource(image)
    }


    fun setCloseButtonHidden(hidden: Boolean) {
        mCloseButton.visibility = if (hidden) View.INVISIBLE else View.VISIBLE
    }

    fun setOnCloseButtonClickListener(listener: (v: View?) -> Unit) {
        mCloseButton.setOnClickListener(object :View.OnClickListener{
            override fun onClick(v: View?) {
                listener(v)
            }

        })
    }
}
